internal/pkg/term/syncbuffer/termprinter.go

```diff
@@ -0,0 +1,176 @@
+func (ltp *LabeledTermPrinter) IsDone() bool {
+	for _, buf := range ltp.buffers {
+		if !buf.IsDone() {
+			return false
+		}
+	}
+	return true
+}
+
+// Print prints the label and the last N lines of logs from each buffer
+// to the LabeledTermPrinter fileWriter and erases the previous output.
+// If numLines is -1 then print all the values from buffers.
+func (ltp *LabeledTermPrinter) Print() error {
+	if ltp.numLines == printAllLinesInBuf {
+		ltp.printAll()
+		return nil
+	}
+	if ltp.prevWrittenLines > 0 {
+		cursor.EraseLinesAbove(ltp.term, ltp.prevWrittenLines)
+	}
+	ltp.prevWrittenLines = 0
+	for _, buf := range ltp.buffers {
+		logs := buf.lines()
+		if len(logs) == 0 {
+			continue
+		}
+		outputLogs := ltp.lastNLines(logs)
+		ltp.writeLines(buf.label, outputLogs)
+		writtenLines, err := ltp.calculateLinesCount(buf.label, outputLogs)
+		if err != nil {
+			return fmt.Errorf("get terminal width: %w", err)
+		}
+		ltp.prevWrittenLines += writtenLines
+	}
+	return nil
+}
+
+// printAll writes the entire contents of all the buffers to the file writer.
+// If one of the buffer gets done then print entire content of the buffer.
+// Until all the buffers are written to file writer.
+func (ltp *LabeledTermPrinter) printAll() {
+	for idx, buf := range ltp.buffers {
+		if !buf.IsDone() {
+			continue
+		}
+		outputLogs := ltp.buffers[idx].lines()
+		ltp.writeLines(buf.label, outputLogs)
+	}
+}
+
+// lastNLines returns the last N lines of the given logs where N is the value of tp.numLines.
+// If the logs slice contains fewer than N lines, all lines are returned.
+// If the given input logs are empty then return slice of empty strings.
+func (ltp *LabeledTermPrinter) lastNLines(logs []string) []string {
+	var start int
+	if len(logs) > ltp.numLines {
+		start = len(logs) - ltp.numLines
+	}
+	end := len(logs)
+
+	// Extract the last N lines
+	logLines := make([]string, ltp.numLines)
+	idx := 0
+	for start < end {
+		logLines[idx] = logs[start]
+		start++
+		idx++
+	}
+	return logLines
+}
+
+// writeLines writes a label and output logs to the terminal associated with the TermPrinter.
+func (ltp *LabeledTermPrinter) writeLines(label string, outputLogs []string) {
+	fmt.Fprintln(ltp.term, label)
+	for _, logLine := range outputLogs {
+		fmt.Fprintf(ltp.term, "%s%s\n", strings.Repeat(" ", ltp.padding), logLine)
+	}
+}
+
+// calculateLinesCount returns the number of lines needed to print the given string slice based on the terminal width
+// or an error when failed to fetch terminal width.
+func (ltp *LabeledTermPrinter) calculateLinesCount(label string, lines []string) (int, error) {
+	width, err := ltp.terminalWidth(ltp.term)
+	if err != nil {
+		return 0, fmt.Errorf("get terminal width: %w", err)
+	}
+	numLines := float64(len(label))
+	for _, line := range lines {
+		// Empty line should be considered as a new line
+		if line == "" {
+			numLines += 1
+		}
+		numLines += math.Ceil(float64(len(line)) + float64(ltp.padding)/float64(width))
```

Comment:
i think the order of operations here is wrong - it looks like `len(line) + ltp.padding/width`, but should be `(len(line) + ltp.padding))/width)`. also, we aren't factoring in the width on the label line 🤔 

i think we can make this more consistent - and only loop through lines once - if we change `writeLines` to output the number of lines written. something like:
```go
writeLines() (int, error) {  
  width, err := termWidth()
  // handle err

  numLines := 0
  writeLine := func(line string) {
    fmt.Fprintln(ltp.term, line)
    if len(line) == 0 {
      numLines++
      return
    }
    numLines += math.Ceil(float64(len(line))/float64(width))
  }

  writeLine(ltp.label)
  for _, line := range lines {
    writeLine(fmt.Sprintf("%s%s", strings.Repeat("", ltp.padding), line))
  }
}
```

Reply:
Done!